### PR TITLE
Add an ability to return a duplicated outputs from the DALI pipeline

### PR DESCRIPTION
**Pull request by JanuszL**
- adds a code that prevents adding "MakeContiguous" multiple times to the output
- adds tests for CPU, GPU and Mixed operators outputs that duplicates
- some small clean up in used API checker in the Pipeline

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
- add an ability to return a duplicated outputs from the DALI pipeline

#### What happened in this PR?
 - adds a code that prevents adding "MakeContiguous" multiple times to the output
 - adds a visible deprecation warning for some pipeline methods
 - some small code cleanup
 - tests are added for the duplicated outputs (CPU, GPU and Mixed operators outputs)
 - no docs update is necessary

**JIRA TASK**: [NA]

### CODE REVIEW

File: dali/python/nvidia/dali/pipeline.py [hunk 1]
```diff
@@ -247,6 +256,19 @@ def _prepare_graph(self):
...
+        output_map = {}
+        next_free_idx = 0
+        for name in outputs:
+            idx = output_map.get(name, next_free_idx + 1)
+            if idx > next_free_idx:
+                output_map[name] = next_free_idx
+                idx = next_free_idx
+                next_free_idx += 1
+                unique_outputs.append(name)
+            self._outputs_list.append(idx)
```
Comment (klecki) [inline dali/python/nvidia/dali/pipeline.py]: I wanted to suggest something that is clearer when indexing, but it does lookup the dictionary twice.
```
for name in outputs:
    if name not in output_map:
        output_map[name] = next_free_idx
        next_free_idx +=1
        unique_outputs.append(name)
     self._outputs_list.append(output_map[name])
```

File: dali/python/nvidia/dali/pipeline.py [hunk 2]
```diff
@@ -19,8 +19,16 @@
 from nvidia.dali import types
 from nvidia.dali import check_edge as Edge
 from threading import local as tls
+import warnings
 pipeline_tls = tls()
 
+def show_deprecation_warning(deprecated, in_favor_of):
```
Comment (klecki) [inline dali/python/nvidia/dali/pipeline.py]:
```suggestion
def _show_deprecation_warning(deprecated, in_favor_of):
```

Comment (JanuszL) [reply dali/python/nvidia/dali/pipeline.py]: Done

File: dali/python/nvidia/dali/pipeline.py (hunk 1 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/pipeline.py]: It is not perf critical. Done.

File: dali/pipeline/pipeline.cc [hunk 3]
```diff
@@ -476,23 +478,27 @@ void Pipeline::Build(vector<std::pair<string, string>> output_names) {
...
+        if (unique_outputs.find(name + device) == unique_outputs.end()) {
+          // Add a make contiguous op to produce this output
+          OpSpec spec =
+            OpSpec("MakeContiguous")
+            .AddArg("device", "mixed")
+            .AddInput(name, "cpu")
+            .AddOutput("contiguous_" + name, "cpu");
+          PrepareOpSpec(&spec, GetNextInternalLogicalId());
+
+          graph_.AddOp(spec, "__MakeContiguous_" + name);
```
Comment (mzient) [inline dali/pipeline/pipeline.cc]:
```suggestion
          graph_.AddOp(spec, "__MakeContiguous_" + name);
          it->second.has_contiguous = true;
```
This should solve the issue without the need for an extra `unordererd_set` - and I think it's missing anyway.

Comment (JanuszL) [reply dali/pipeline/pipeline.cc]: Done

**Pull request merged**